docs/publishing/index.md

```diff
@@ -0,0 +1,125 @@
+
+## Do import 3rd party node modules with "bare" import specifiers
+
+When referring to 3rd party dependencies in your code, you cannot guarantee that implementing projects rely on the same file structure as you do. To prevent dependency pointing from causing issue across different environments be sure to import these dependencies into your project with "bare" import specifiers so that tooling in your users' environment can manage resolving the location of those dependencies; e.g. `import { LitElement } from 'lit-element';`
+
+## Do include file extensions in import specifiers
+
+When importing something that doesn't point directly to the entry point of a module (e.g.: `import {LitElement} from 'lit-element';`) always make sure to include a file extension (e.g. `import { ifDefined } from 'lit-html/directives/if-defined.js';`).
+
+## Optional: use `type: "module"`
+
+Using `type: "module"` in your `package.json` is used to help CDNs and bundlers choose whether to parse your package with module parse goal. However, it will also enforce strict realities on how es modules and common JS modules must be imported into your scripting. If all of the JS in your package is es modules, and using this entry in your `package.json` does not cause issues with your scripting, then this is a great way to support tooling the references your package in leveraging it more efficiently. If scripting in your package has issues running in Node@12+ due to this inclusion, excluding it will ease the local requirements on import statements in your code.
+
+## Optional: include typings
+
+`*.d.ts` files are an important bridge between your code and users that choose to develop in TypeScript. If you are already developing in TypeScript, then generating these is as simple as adding the appropriate property to your `tsconfig.json`. If you are developing in JS, it can still be useful to include typings; learn more about [generating type definitions from javascript](https://dev.to/open-wc/generating-typescript-definition-files-from-javascript-5bp2).
+
+## Optional: include source maps
+
+If you choose work in TypeScript, or to use syntax that is newer than the standards discussed above, it can be important to include source maps and the source files (TypeScript or otherwise) that your published code originates from to make understanding that code, as well as possibly reporting/fixing issues in that code, easier for your users.
+
+## Do not optimize
+
+Building is an application level concern. It's not repeating yourself if it is a really important point, it's merely ensuring that it sinks in. Your package will be leveraged by a developer that will know more about their specific delivery targets that you will and it is important that you don't attempt to optimize your code beyond delivering the best possible standard EcmaScript that you can.
```

Comment:
```suggestion
Bundling is an application level concern. It's not repeating yourself if it is a really important point, it's merely ensuring that it sinks in. Your package will be leveraged by a developer that will know more about their specific delivery targets that you will and it is important that you don't attempt to optimize your code beyond delivering the best possible standard EcmaScript that you can.
```

Reply:
I've gone back and forth on this word... does "bundling" really include translating, minifying, optimizing, merging modules together, and structuring it for production? Might be thinking too hard, but it seems exclusive only to be inclusive of other things later.

Reply:
> does "bundling" really include translating, minifying, optimizing, merging modules together, and structuring it for production?

IMO; yes. But I see what you're saying. Really what we want to say is "minifying, optimizing, transpiling, etc is an application level concern". 

So... I'm not really sure how to go about this one 😅 Maybe get some feedback from others on this?

Reply:
I think I have a good idea here, new version coming soon.